src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -1317,6 +1318,173 @@ static class C
     }
 
+    [Fact]
+    public void Conversion_Array_Span_Implicit_MethodGroup_ExtensionMethodReceiver_Inferred()
+    {
+        var source = """
+            using System;
+
+            var a = new[] { 1, 2, 3 };
+            var d1 = a.M;
+            var d2 = x => a.M(x);
+            var d3 = (int x) => a.M(x);
+
+            static class C
+            {
+                public static int M(this Span<int> x, int y) => x[y];
+            }
+            """;
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (4,10): error CS8917: The delegate type could not be inferred.
+            // var d1 = a.M;
+            Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "a.M").WithLocation(4, 10),
+            // (5,10): error CS8917: The delegate type could not be inferred.
+            // var d2 = x => a.M(x);
+            Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "x => a.M(x)").WithLocation(5, 10),
+            // (6,21): error CS1929: 'int[]' does not contain a definition for 'M' and the best extension method overload 'C.M(Span<int>, int)' requires a receiver of type 'System.Span<int>'
+            // var d3 = (int x) => a.M(x);
+            Diagnostic(ErrorCode.ERR_BadInstanceArgType, "a").WithArguments("int[]", "M", "C.M(System.Span<int>, int)", "System.Span<int>").WithLocation(6, 21));
+
+        var expectedDiagnostics = new[]
+        {
+            // (4,10): error CS1113: Extension method 'C.M(Span<int>, int)' defined on value type 'Span<int>' cannot be used to create delegates
+            // var d1 = a.M;
+            Diagnostic(ErrorCode.ERR_ValueTypeExtDelegate, "a.M").WithArguments("C.M(System.Span<int>, int)", "System.Span<int>").WithLocation(4, 10),
+            // (5,10): error CS8917: The delegate type could not be inferred.
+            // var d2 = x => a.M(x);
+            Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "x => a.M(x)").WithLocation(5, 10)
+        };
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+        var comp = CreateCompilationWithSpan(source).VerifyDiagnostics(expectedDiagnostics);
+
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+        var aVariable = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().First();
+        var aSymbol = (ILocalSymbol)model.GetDeclaredSymbol(aVariable)!;
+        AssertEx.Equal("System.Int32[]", aSymbol.Type.ToTestDisplayString());
+        var d1Access = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>()
+            .First(s => s.Expression.ToString() == "a");
+        var lookupResult = model.LookupSymbols(d1Access.Name.SpanStart, aSymbol.Type, "M", includeReducedExtensionMethods: true);
+        AssertEx.Equal("System.Int32 System.Span<System.Int32>.M(System.Int32 y)", lookupResult.Single().ToTestDisplayString());
```

Comment:
Without the product change
- the error messages would be that the extension method is not applicable (CS1929) rather than that Span cannot be a delegate receiver (CS1113) - but that's just different diagnostics, so probably not very interesting
- however also this LookupSymbols would not be able to find the method

Reply:
FWIW, it looks like we'll report "error CS8917: The delegate type could not be inferred" rather than "error CS1113: Extension method ... cannot be used to create delegates" if the `ReducedExtensionMethodSymbol` change is removed.